app/controllers/submissions_controller.rb

```diff
@@ -11,9 +11,8 @@ def create
     @task.update status: :waiting if @task.not_submitted? || @task.waiting?
     @submission = Submission.create(submission_params)
     UserMailer.new_submission_notify(@submission).deliver
+    Notification.new_event(user: @submission.teacher, task: @submission.task, event_type: :new_submission, event_time: @submission.created_at)
```

Comment:
Hm. I see, there's a condition on a task update. You can explicitly [touch](http://apidock.com/rails/ActiveRecord/Timestamp/touch) a task.